docs/pages/api-reference/liveblocks-react.mdx

```diff
@@ -10,21 +10,1134 @@ alwaysShowAllNavigationLevels: false
+    Presence resets every time they disconnect. This object must be
+    JSON-serializable. This value is ignored after the first render. [Learn
+    more](#setting-initial-presence).
+  </PropertiesListItem>
+  <PropertiesListItem name="initialStorage" type="LsonObject">
+    The initial Storage structure for the room when it’s joined for the first
+    time. This is only set a single time, when the room has not yet been
+    populated. This object must contain [conflict-free live
+    structures](/docs/api-reference/liveblocks-client#Storage). This value is
+    ignored after the first render, and if Storage for the current room has
+    already been created. [Learn more](#setting-initial-storage).
+  </PropertiesListItem>
+  <PropertiesListItem name="autoConnect" type="boolean" defaultValue="true">
+    Whether the room immediately connects to Liveblocks servers. This value is
+    ignored after the first render.
+  </PropertiesListItem>
+</PropertiesList>
+
+#### Setting initial Presence [#setting-initial-presence]
+
+Presence is used for storing temporary user-based values, such as a user’s
+cursor coordinates, or their current selection. Each user has their own
+presence, and this is readable for all other connected users. Set your initial
+Presence value by using `initialPresence`.
+
+```tsx highlight="7-12"
+import { RoomProvider } from "@liveblocks/react/suspense";
+
+function App() {
+  return (
+    <RoomProvider
+      id="my-room"
+      initialPresence={{
+        cursor: null,
+        colors: ["red", "purple"],
+        selection: {
+          id: 72426,
+        }}
+    >
+      {/* children */}
+    </RoomProvider>
+  );
+}
+```
+
+Each user’s Presence resets every time they disconnect, as this is only meant
+for temporary data. Any JSON-serializable object is allowed (the `JsonObject`
+type).
+
+#### Setting initial Storage [#setting-initial-storage]
+
+Storage is used to store permanent data that’s used in your application, such as
+shapes on a whiteboard, nodes on a flowchart, or text in a form. The first time
+a room is entered, you can set an initial value by using `initialStorage`. Note
+that this value is only read a single time.
+
+```tsx highlight="9-15"
+import { LiveList, LiveObject } from "@liveblocks/client";
+import { RoomProvider } from "@liveblocks/react/suspense";
+
+function App() {
+  return (
+    <RoomProvider
+      id="my-room"
+      initialPresence={}
+      initialStorage={{
+        title: "Untitled",
+        names: new LiveList(["Steven", "Guillaume"])
+        shapes: new LiveMap([
+          ["g9shu0", new LiveObject({ type: "rectangle", color: "red" })],
+          ["djs3g5", new LiveObject({ type: "circle", color: "yellow" })],
+        ]),
+      }}
+    >
+      {/* children */}
+    </RoomProvider>
+  );
+}
+```
+
+Any
+[conflict-free live structures](/docs/api-reference/liveblocks-client#Storage)
+and JSON-serializable objects are allowed (the `LsonObject` type).
+
 ### createRoomContext
```

Comment:
Maybe we could make a separate page in the docs on the topic of "looks like I can only set one global room type, but my app needs multiple room types", and link there?

Reply:
Good idea! It's on my to-do list for after the launch LB-852.